org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ClasspathJar.java

```diff
@@ -125,13 +125,18 @@ IModule initializeModule() {
 	ZipFile file = null;
 	try {
 		file = new ZipFile(this.zipFilename);
-		String releasePath = "META-INF/versions/" + this.compliance + '/' + IModule.MODULE_INFO_CLASS; //$NON-NLS-1$
 		ClassFileReader classfile = null;
-		try {
-			classfile = ClassFileReader.read(file, releasePath);
-		} catch (Exception e) {
-			e.printStackTrace();
-			// move on to the default
+		for (int version = Integer.parseInt(this.compliance); version >= 9; version--) {
```

Comment:
and here, we search for the `module-info.class`, which should be the fix for the bug

Reply:
Any reason why It's being done in ClasspathJar instead of org.eclipse.jdt.internal.core.builder.ClasspathMultiReleaseJar ?

Reply:
@jarthana Do you mean, why I have made this change or why this code is here at all?

IMHO it sounds logical, if this is handled in ClasspathMultiReleaseJar as it does not require to have duplicate code.
(But I'm really not the expert to do such refactoring in this project)

Reply:
I was just wondering about having this in the super class ClasspathJar but in case of the compiler.batch package, you have made it in the subclass ClasspathMultiReleaseJar. Haven't looked at the code in detail, but IIRC, we had similar code and hierarchy in both the compiler.batch and core.builder packages.